orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/StartWaitingExecutionsHandler.kt

```diff
@@ -61,12 +65,21 @@ class StartWaitingExecutionsHandler(
       // when not purging the queue, run the messages in the order they came in
       pendingExecutionService.popOldest(message.pipelineConfigId)
     }
-      ?.let {
-        // spoiler, it always is!
-        if (it is ExecutionLevel) {
-          log.info("Starting queued pipeline {} {} {}", it.application, it.executionId)
-        }
-        queue.push(it)
+
+    if (pendingMessage != null) {
+
+      // spoiler, it always is!
+      if (pendingMessage is ExecutionLevel) {
+        log.info("Starting queued pipeline {} {} {}", pendingMessage.application, pendingMessage.executionId)
```

Comment:
I know it's not yours but isn't this weird? It always pushes `pendingMessage`, but _only_ logs if it is `ExecutionLevel` 🤔

`queue.push(pendingMessage)` probably belongs in the body of the if statement

Reply:
it is weird... despite themessage always being executionlevel... i am scared of changing this :)